### PR TITLE
[5.0] Improve ext-mbstring Requirements In composer.json Files

### PR DESCRIPTION
**Pull request by vlakoff**
Thankfully, a simple search for "mb_" in files does the job.

By the way, Stringy compatibility has been improved (refs https://github.com/danielstjules/Stringy/pull/57).

### CODE REVIEW

File: src/Illuminate/Filesystem/composer.json [hunk 1]
```diff
@@ -15,6 +15,9 @@
         "symfony/finder": "2.6.*",
         "league/flysystem": "~1.0"
     },
+    "require-dev": {
```
Comment (GrahamCampbell) [inline src/Illuminate/Filesystem/composer.json]: please remove this

**Comment by GrahamCampbell** [comment]
I'm a little confused by these changes. Also, you haven't updated the support package.

**Comment by vlakoff** [comment]
Already present in support package.

For the rest, I think it's mainly about modularity vs (maybe) maintainability.
I gave you the elements, now I let you decide :)

**Comment by vlakoff** [comment]
The require-*dev* for mbstring in Filesystem package could easily be removed. Only [one line](https://github.com/laravel/framework/blob/44623342bc3907eab43860eb2f0cacdc59ef09fa/tests/Filesystem/FilesystemTest.php#L174) (recently introduced in #6905) needs mbstring, and is easy to make work without mbstring. It's just that it would be conflicting with this PR.

File: composer.json [hunk 2]
```diff
@@ -13,9 +13,8 @@
         "php": ">=5.4.0",
         "ext-openssl": "*",
         "ext-mcrypt": "*",
-        "ext-mbstring": "*",
```
Comment (GrahamCampbell) [inline composer.json]: keep this if you're not removing it in the support package

Comment (vlakoff) [inline composer.json]: I don't get it... Support package definitely needs the ext, why would I remove it? Then, isn't the ext redundant in the main package?

Comment (GrahamCampbell) [inline composer.json]: No. The main composer.json doesn't inherit the others. It just totally replaces them. They aren't even looked at by composer.

Comment (vlakoff) [inline composer.json]: Ahh, right. Fixing.

File: src/Illuminate/Auth/composer.json [hunk 3]
```diff
@@ -10,6 +10,7 @@
     ],
     "require": {
         "php": ">=5.4.0",
+        "ext-mbstring": "*",
```
Comment (GrahamCampbell) [inline src/Illuminate/Auth/composer.json]: please remove this

File: src/Illuminate/Filesystem/composer.json [hunk 4]
```diff
@@ -15,6 +15,9 @@
...
+        "ext-mbstring": "*",
```
Comment (GrahamCampbell) [inline src/Illuminate/Filesystem/composer.json]: please remove this

**Comment by GrahamCampbell** [comment]
This makes no sense.

**Comment by GrahamCampbell** [comment]
If support needs the extension then so do the others. They inherit the dependency. It doesn't need adding again.

**Comment by vlakoff** [comment]
I was a bit in a hurry and missed that obvious point. My apologies. So you can simply update both stringy requirements (just to be on the safe side) and everything's fine.

**Pull request closed without merging**